src/main/scala/performanceanalysis/LogParserActor.scala

```diff
@@ -22,85 +24,77 @@ class LogParserActor extends Actor with ActorLogging {
   this: AlertRuleActorCreator =>
   private type AlertRuleActorRef = ActorRef
 
-  def receive: Receive = normal(Nil, List.empty, Map())
+  private val metrics = mutable.MutableList[Metric]()
+  private val logLines = mutable.MutableList[String]()
+  private val alertsByMetricKey = MutableMap[MetricKey, List[AlertRuleActorRef]]()
 
-  def normal(metrics: List[Metric],
-             logLines: List[String],
-             alertsByMetricKey: Map[MetricKey, List[AlertRuleActorRef]]): Receive = {
+  def receive: Receive = {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Cyclomatic complexity of 11 exceeds max of 10](https://www.codacy.com/app/scala-academy/performance-analysis/file/2671056344/issues/source?bid=3220915&fileBranchId=3331880#l31)